File: ios/Records/Slave.swift
import ExpoModulesCore

struct Slave: Record, Equatable {
    @Field
    var source: String = ""

    @Field
    var type: String = ""

    @Field
    var selected: Bool? = false

    // swiftformat:disable:next redundantEquatable
    static func == (lhs: Slave, rhs: Slave) -> Bool {
        lhs.source == rhs.source &&
            lhs.type == rhs.type &&
            lhs.selected == rhs.selected
    }
}

File: android/src/main/java/expo/modules/libvlcplayer/records/Slave.kt
package expo.modules.libvlcplayer.records

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record
import java.io.Serializable

class Slave(
    @Field var source: String = "",
    @Field var type: String = "",
    @Field var selected: Boolean? = false,
) : Record,
    Serializable
